libs/blocks/global-navigation/blocks/navDropdown/dropdown.js

```diff
@@ -0,0 +1,275 @@
+  elem.remove();
+
+  return headline;
+};
+
+const decorateLinkGroup = (elem, index) => {
+  if (!(elem instanceof HTMLElement) || !elem.querySelector('a')) return null;
+
+  // TODO: allow links with image and no label
+  const image = elem.querySelector('picture');
+  const link = elem.querySelector('a');
+  const description = elem.querySelector('p:nth-child(2)');
+
+  const imageElem = image ? toFragment`<div class="feds-navLink-image">${image}</div>` : '';
+  const descriptionElem = description ? toFragment`<div class="feds-navLink-description">${description.textContent}</div>` : '';
+  const contentElem = link ? toFragment`<div class="feds-navLink-content">
+      <div class="feds-navLink-title">${link.textContent}</div>
+      ${descriptionElem}
+    </div>` : '';
+  const linkGroup = toFragment`<a 
+    href="${localizeLink(link.href)}"
+    class="feds-navLink"
+    daa-ll="${getAnalyticsValue(link.textContent, index)}">
+      ${imageElem}
+      ${contentElem}
+    </a>`;
+
+  return linkGroup;
+};
+
+// Decorate a list of simple links
+const decorateLinkList = ({ list, className = 'feds-navLink', includeCta = true } = {}) => {
+  list.querySelectorAll('a').forEach((link, index) => {
+    // If the link is wrapped in a 'strong' or 'em' tag, make it a CTA
+    if (includeCta
+      && (link.parentElement.tagName === 'STRONG' || link.parentElement.tagName === 'EM')) {
+      const type = link.parentElement.tagName === 'EM' ? 'secondaryCta' : 'primaryCta';
+      decorateCta({ elem: link, type, index: index + 1 });
+    // Otherwise add analytics attributes and proper class
+    } else {
+      link.setAttribute('daa-ll', getAnalyticsValue(link.textContent, index + 1));
+      link.classList.add(className);
+    }
+  });
+};
+
+// Current limitation: we can only add one link
+const decoratePromo = (elem) => {
+  const isDarkTheme = elem.classList.contains('dark');
+  const isImageOnly = elem.classList.contains('image-only');
+  const imageElem = elem.querySelector('picture');
+
+  decorateLinkList({ list: elem, className: 'feds-promo-link', includeCta: false });
+
+  const decorateImage = () => {
+    const linkElem = elem.querySelector('a');
+    const imageWrapper = imageElem.closest('.gnav-promo > div');
+    let promoImageElem;
+
+    if (linkElem instanceof HTMLElement) {
+      promoImageElem = toFragment`<a class="feds-promo-image" href="${localizeLink(linkElem.href)}">
+          ${imageElem}
+        </a>`;
+    } else {
+      // TODO: is there really any use-case where a promo doesn't have a link?
+      promoImageElem = toFragment`<div class="feds-promo-image">
+          ${imageElem}
+        </div>`;
+    }
+
+    // If the promo is set to 'image-only',
+    // replace the whole promo content with the decorated image
+    if (isImageOnly) {
+      elem.replaceChildren(promoImageElem);
+    } else {
+    // Otherwise, just replace the image container with the decorated image
+      imageWrapper.replaceWith(promoImageElem);
```

Comment:
I'll make the change to check for the element. However, I can't move the declaration of `imageWrapper`, since we need a reference to its initial state. Until the logic reaches this point, it will have been moved to another parent.